src/spyglass/utils/dj_mixin.py

```diff
@@ -920,6 +943,47 @@ def check_threads(self, detailed=False, all_threads=False) -> DataFrame:
 
         return df
 
+    # --------------------------- Check disc usage ------------------------------
+    def get_table_storage_usage(self, human_readable=False):
+        """Total size of all analysis files in the table.
+        Uses the analysis_file_name field to find the file paths and sum their
+        sizes.
+        Parameters
+        ----------
+        human_readable : bool, optional
+            If True, return a human-readable string of the total size.
+            Default False, returns total size in bytes.
+
+        Returns
+        -------
+        Union[str, int]
+            Total size of all analysis files in the table. If human_readable is
+            True, returns a string with the size in bytes, KiB, MiB, GiB, TiB,
+            or PiB. If human_readable is False, returns the total size in bytes.
+
+        """
+        if "analysis_file_name" not in self.heading.names:
+            logger.warning(
+                f"{self.full_table_name} does not have an analysis_file_name field."
+            )
+            return "0 Mib" if human_readable else 0
+        file_names = self.fetch("analysis_file_name")
+        file_paths = [
+            _quick_get_analysis_path(file_name) for file_name in file_names
+        ]
+        file_paths = [path for path in file_paths if path is not None]
+        file_sizes = [os.stat(path).st_size for path in file_paths]
+        total_size = sum(file_sizes)
+        if not human_readable:
+            return total_size
+        human_size = total_size
+        for unit in ["B", "KiB", "MiB", "GiB", "TiB", "PiB"]:
+            if human_size < 1024.0 or unit == "PiB":
+                break
+            human_size /= 1024.0
+        human_size = f"{human_size:.2f} {unit}"
+        return human_size
```

Comment:
Please extract this to a util. I would then edit [my PR](https://github.com/CBroz1/spyglass/blob/frc/src/spyglass/utils/nwb_hash.py#L21) accordingly